mongo-long-stream-test/src/main/java/com/hazelcast/jet/tests/mongo/stream/MongoLongStreamTest.java

```diff
@@ -0,0 +1,197 @@
+            }
+            SECONDS.sleep(1);
+        }
+        final long actualTotalCount = latestCounterMap.get(clusterName);
+        assertEquals(expectedTotalCount, actualTotalCount);
+    }
+
+    private static void log(final ILogger logger, final String message, final String clusterName) {
+        logger.info("Cluster" + clusterName + "\t\t" + message);
+    }
+
+    @Override
+    public void init(final HazelcastInstance client) {
+        mongoConnectionString = "mongodb://" + property("mongoIp", "127.0.0.1") + ":27017";
+        snapshotIntervalMs = propertyInt("snapshotIntervalMs", DEFAULT_SNAPSHOT_INTERVAL);
+    }
+
+    @Override
+    protected boolean runOnBothClusters() {
+        return true;
+    }
+
+    @Override
+    public void test(final HazelcastInstance client, final String clusterName) throws Exception {
+        final long begin = System.currentTimeMillis();
+
+        deleteCollectionAndCreateNewOne(clusterName);
+
+        final StreamSource<Document> mongoSource = MongoSources
+                .stream(clusterName + "_mongo-long-stream", getMongoClient(mongoConnectionString))
+                .database(MONGO_DATABASE)
+                .collection(clusterName)
+                .startAtOperationTime(MongoUtilities.bsonTimestampFromTimeMillis(begin))
+                .build();
+
+        final Pipeline fromMongo = Pipeline.create();
+        fromMongo.readFrom(mongoSource)
+                .withNativeTimestamps(0)
+                .map(doc -> doc.getLong("docId"))
+                .writeTo(VerificationProcessor.sink(clusterName));
+
+        final JobConfig jobConfig = new JobConfig();
+
+        if (clusterName.startsWith(DYNAMIC_CLUSTER)) {
+            jobConfig.setSnapshotIntervalMillis(snapshotIntervalMs);
+            jobConfig.setProcessingGuarantee(EXACTLY_ONCE);
+        } else {
+            jobConfig.addClass(MongoLongStreamTest.class, MongoDocsProducer.class,
+                    VerificationProcessor.class, MongoClientSupplier.class);
+        }
+
+        jobConfig.setName(clusterName + "_" + MONGO_DATABASE);
+
+        final Job job = client.getJet().newJob(fromMongo, jobConfig);
+        assertJobStatusEventually(job);
+
+        final MongoDocsProducer producer = new MongoDocsProducer(mongoConnectionString,
+                MONGO_DATABASE,
+                clusterName,
+                logger);
+        producer.start();
+
+        final long expectedTotalCount;
+        long lastlyProcessed = -1;
+        try {
+            while (System.currentTimeMillis() - begin < durationInMillis) {
+                if (getJobStatusWithRetry(job) == FAILED) {
+                    job.join();
+                } else {
+                    final long currentlyProcessedDocs = getNumberOfCurrentlyProcessedDocs(client, clusterName);
+
+                    if (currentlyProcessedDocs == lastlyProcessed) {
+                        log(logger, "Nothing was processed in last 1 minute, current counter:"
```

Comment:
We should fail the test if nothing new is not process for 5 or 10 minutes. It's really easy to miss that nothing is processed if we just log it.